sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -317,32 +428,54 @@ def _environments_query(
             return query.lock(copy=False)
         return query
 
+    def _hydrate_seeds(self, snapshot_ids: t.Optional[t.Iterable[SnapshotIdLike]] = None) -> None:
+        query = (
+            exp.select("name", "identifier", "content")
+            .from_(self.seeds_table)
+            .where(None if not snapshot_ids else self._snapshot_id_filter(snapshot_ids))
+        )
+        for name, identifier, content in self.engine_adapter.fetchall(query):
+            snapshot = self._snapshot_cache.get(SnapshotId(name=name, identifier=identifier))
+            if snapshot and not snapshot.seed_is_hydrated:
```

Comment:
Should we check whether the cached version is hydrated prior to fetching its contents?